SharpBucket/Utility/StringExtensions.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace SharpBucket.Utility
+{
+    public static class StringExtensions
+    {
+        /// <summary>
+        /// Converts a repository name into the corresponding slug.
+        /// If a UUID in B or D format is passed in, it will be returned in B format.
+        /// </summary>
+        /// <param name="repoName">The name of a repository.</param>
+        /// <returns>The slug version of the repository name, or the formatted UUID.</returns>
+        public static string ToSlug(this string repoName)
+        {
+            if (repoName.TryGetGuid(out var guidString))
+                return guidString;
+
+            if (repoName.Contains('\''))
```

Comment:
Well, `Replace()` is an `extern` call, so I would imagine there's some overhead for calling out to what looks like unmanaged code and therefore a slight benefit to checking `Contains()` first. But since I'm not calling `IsMatch()` before the regex `Replace()`, I may as well take out this check as well.